homeassistant/components/ping/coordinator.py

```diff
@@ -51,8 +55,28 @@ def __init__(
         )
 
     async def _async_update_data(self) -> PingResult:
-        """Trigger ping check."""
-        await self.ping.async_update()
+        """Update ping data within a timeout duration."""
+
+        if self._update_task is not None:
+            if not self._update_task.done():
+                # don't queue another task
+                return self._latest_ping_result()
+        else:
```

Comment:
```suggestion
        if self._update_task is None:
```

if the task is not done yet, should we `wait_for` it instead of immediately returning old results? It may complete on the time window for this call to `_async_update_data`.